<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyboardLayoutPropertiesWidget</class>
 <widget class="QWidget" name="KeyboardLayoutPropertiesWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>562</width>
    <height>136</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_6">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QStackedWidget" name="m_stackedWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="m_keyboardProperties">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="m_keyboardGroupBox">
         <property name="title">
          <string>Keyboard</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>15</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <property name="verticalSpacing">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="m_keyboardLayouttTitleLabel">
              <property name="text">
               <string>Title:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyboardLayoutTitleLineEdit</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="KLineEdit" name="m_keyboardLayoutTitleLineEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="m_keyboardLayoutNameLabel">
              <property name="text">
               <string>Name:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyboardLayoutNameEdit</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="KLineEdit" name="m_keyboardLayoutNameEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="formLayout_2">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <property name="verticalSpacing">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="m_keyboardLayoutWidthLabel">
              <property name="text">
               <string>Width:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyboardLayoutWidthSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="m_keyboardLayoutWidthSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>100</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="m_keyboardLayoutHeightLabel">
              <property name="text">
               <string>Height:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyboardLayoutHeightSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="m_keyboardLayoutHeightSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>100</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="m_keyProperties">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Geometry</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="spacing">
           <number>15</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout_3">
            <property name="horizontalSpacing">
             <number>0</number>
            </property>
            <property name="verticalSpacing">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Left:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyLeftSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Top:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyTopSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="m_keyLeftSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="m_keyTopSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="formLayout_4">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <property name="horizontalSpacing">
             <number>0</number>
            </property>
            <property name="verticalSpacing">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Width:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyWidthSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="m_keyWidthSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>20</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Height:</string>
              </property>
              <property name="buddy">
               <cstring>m_keyHeightSpinBox</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="m_keyHeightSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>20</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QStackedWidget" name="m_subStackedWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>2</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="m_keyPropertiesSubWidget">
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QGroupBox" name="groupBox_5">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string>Key</string>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
             <layout class="QFormLayout" name="formLayout_7">
              <property name="fieldGrowthPolicy">
               <enum>QFormLayout::ExpandingFieldsGrow</enum>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="m_keyFingerLabel">
                <property name="text">
                 <string>Finger:</string>
                </property>
                <property name="buddy">
                 <cstring>m_keyFingerComboBox</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="KComboBox" name="m_keyFingerComboBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <item>
                 <property name="text">
                  <string>Left little finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Left ring finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Left middle finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Left index finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Right index finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Right middle finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Right ring finger</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Right little finger</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="m_keyHapticMarkerLabel">
                <property name="text">
                 <string>Haptic Marker:</string>
                </property>
                <property name="buddy">
                 <cstring>m_keyHapticMarkerCheckBox</cstring>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QCheckBox" name="m_keyHapticMarkerCheckBox">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string>Characters</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QTableView" name="m_charactersView">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="selectionMode">
                 <enum>QAbstractItemView::SingleSelection</enum>
                </property>
                <property name="selectionBehavior">
                 <enum>QAbstractItemView::SelectRows</enum>
                </property>
                <attribute name="verticalHeaderVisible">
                 <bool>false</bool>
                </attribute>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout">
                <property name="spacing">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QToolButton" name="m_addCharacterButton">
                  <property name="toolTip">
                   <string>Add new character</string>
                  </property>
                  <property name="icon">
                   <iconset theme="list-add"/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="m_removeCharacterButton">
                  <property name="toolTip">
                   <string>Remove character</string>
                  </property>
                  <property name="icon">
                   <iconset theme="list-remove"/>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="m_specialKeyPropertiesSubWidget">
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>Key</string>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
             <layout class="QFormLayout" name="formLayout_5">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="m_specialKeyTypeLabel">
                <property name="text">
                 <string>Type:</string>
                </property>
                <property name="buddy">
                 <cstring>m_specialKeyTypeComboBox</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="KComboBox" name="m_specialKeyTypeComboBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <item>
                 <property name="text">
                  <string>Tab</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Capslock</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Shift</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Backspace</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Return</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Space</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Other</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="m_specialKeyLabelLabel">
                <property name="text">
                 <string>Label:</string>
                </property>
                <property name="buddy">
                 <cstring>m_specialKeyLabelLineEdit</cstring>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="KLineEdit" name="m_specialKeyLabelLineEdit"/>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="m_specialKeyModifierIdLabel">
                <property name="text">
                 <string>Modifier ID:</string>
                </property>
                <property name="buddy">
                 <cstring>m_specialKeyModifierIdLineEdit</cstring>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="KLineEdit" name="m_specialKeyModifierIdLineEdit"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
